### PR TITLE
Let the coach add events without emailing parents

### PR DESCRIPTION
**Pull request by BRCornelius**
## Summary

Adding an event mails every family on the roster (#45), which is right for a new game and
wrong for a game the league already put on everyone's calendar, a placeholder date that is
going to move, or a schedule being rebuilt after a mistake. This adds an "Email parents
about this" checkbox to the add-event form — ticked by default, so the announcing behaviour
is unchanged unless the coach opts out — and a `skipped` announcement outcome the success
banner says out loud.

## Key Decisions

**An unticked checkbox and a form that never had the box submit the same thing: nothing.**
Those two have to mean different things — the second is every POST made before this change
and must still announce. So the form sends a hidden `announce=0` sentinel ahead of the
checkbox's `announce=1`, and `parseAnnounce` reads *absent* as "announce, as always" and a
sentinel-without-checkbox as the skip. Same absent-means-old-behaviour rule `parseRepeat`
already follows for the repeat count. The check is over `getAll("announce")` rather than
`get`, so it does not depend on document order putting the sentinel first.

**"Skipped" is a distinct state from "nobody to tell", and it is not silent.** A past-dated
event already announces nothing and says nothing (`none`), which AGENTS.md notes as a real
cost — a coach who typos the year gets no email and no explanation. That trade does not
carry over here: unticking is a choice, and an accidental untick should be caught on the
page rather than by a receipt that never arrives. So `skipped` gets its own banner clause,
"Parents were not emailed.", in the past tense — unlike `sending`, which is deliberately
present tense because the fan-out has not started when the coach reads it.

**The choice resets to on after every add**, alongside the date, notes and repeat count in
`stickyValues`. Type/location/opponent stay because they barely change between games; this
one is the opposite case — a sticky untick would make the next real game quiet too, and that
failure is silent until a family asks why they never heard. A rejection still hands the
unticked box back with the other typed values, so a mistyped date costs a correction and
does not re-arm the email behind the coach's back.

**The skip happens before anything is resolved**, so a quiet add does not read the roster or
the team at all — nothing about the announcement happens rather than happening and being
discarded.

## What's in Scope

- `announce: boolean` on `EventFormValues`, defaulted on in `EMPTY_EVENT_VALUES` and reset
  to on by `stickyValues`.
- `{status: "skipped"}` added to `AddEventAnnouncement`; `announcementNote` renders it.
- `parseAnnounce` in `actions.ts`; `createEventAction` skips `scheduleAnnouncement` when the
  coach unticked the box.
- The checkbox itself, with its hidden sentinel, a help line, and `aria-describedby` — inside
  the fieldset that locks during a submit, like every other field.
- The duplicate-an-event path passes `announce: true`: it is a property of the submit, not of
  the event being copied.
- AGENTS.md gains the rule beside the existing announcement notes.

### Out of Scope

- Editing an event still sends nothing at all, unchanged — a change-notification needs a diff
  this does not compute.
- No per-team default. The checkbox is per submit; a team-level "never announce" setting would
  be a column and a migration, and the failure mode of a forgotten global default is worse
  than one tap per quiet add.

## Test Plan

- [x] `pnpm check` — lint, typecheck, 2044 tests across 128 files, all passing.
- [x] `pnpm exec next build` — compiles clean (the half `pnpm check` cannot cover, and the
      one that catches a non-async export in a `"use server"` file).
- [ ] Manual: as a coach, go to `/t/<teamId>/schedule`. Confirm "Email parents about this" is
      ticked on the fresh form. Add a future game with it ticked — the banner should read
      "Added Game on … Emailing N parents now — we'll send you a summary." Add another with
      it unticked — the banner should read "… Parents were not emailed." and no mail should
      go out.
- [ ] Manual: after the quiet add, confirm the box is ticked again for the next event, and
      that location/opponent/type are still filled in.
- [ ] Edge cases covered by new tests: field absent entirely (announces, the pre-change
      behaviour); both entries sent (announces); sentinel alone (skips, and never calls
      `listTeamGuardians` or `getTeamById`); entries in either order; a repeat-weekly run
      skipped as one batch; the reset after an add; the unticked box surviving a rejection.

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01QFucsFKru5267YqGLKKHhr

---
_Generated by [Claude Code](https://claude.ai/code/session_01QFucsFKru5267YqGLKKHhr)_

### CODE REVIEW

File: src/app/t/[teamId]/schedule/actions.ts [hunk 1]
```diff
@@ -665,6 +689,15 @@ async function sendReceipt(input: {
  * A validation failure returns too, with what was typed, so a mistyped time
  * costs a correction rather than the whole form.
  *
+ * **The announcement is the coach's to skip, per submit.** "Email parents" is
+ * a checkbox on the form, on by default, and unticking it makes this a plain
+ * write: no roster read, no fan-out, no receipt, and the banner says so. The
+ * cases it exists for are the ones where the email would be noise — a game
+ * the league already put on everyone's calendar, a placeholder date that is
```
[the remark this answers is not in the document]
Comment (BRCornelius) [reply src/app/t/[teamId]/schedule/actions.ts]: Fixed in 90a3ade — you're right, and `listTeamMembers` is exactly the call that made "plain write" wrong. It resolves `coachEmail`, which only the announcement uses, but it sits on the write path ahead of the branch and runs either way.

Both notes now claim only what the branch actually does — skipping `scheduleAnnouncement` and everything downstream (the guardian read, the deferred fan-out, the receipt) — and the doc comment says out loud that the membership lookup still runs, so the narrower claim is the one a later change gets measured against. The inline note at the return site says the same thing in one line.

Comments only, no behaviour change; `pnpm check` still green (2044 tests).

---
_Generated by [Claude Code](https://claude.ai/code)_
